src/httpserver/webserver.hpp

```diff
@@ -167,16 +168,18 @@ class webserver {
      const std::string file_upload_dir;
      const bool generate_random_filename_on_upload;
      const bool deferred_enabled;
-     bool single_resource;
-     bool tcp_nodelay;
+     const bool single_resource;
+     const bool tcp_nodelay;
      pthread_mutex_t mutexwait;
      pthread_cond_t mutexcond;
-     render_ptr not_found_resource;
-     render_ptr method_not_allowed_resource;
-     render_ptr internal_error_resource;
+     const render_ptr not_found_resource;
+     const render_ptr method_not_allowed_resource;
+     const render_ptr internal_error_resource;
+     std::shared_mutex registered_resources_mutex;
      std::map<details::http_endpoint, http_resource*> registered_resources;
      std::map<std::string, http_resource*> registered_resources_str;
 
+     std::shared_mutex bans_and_allowances_mutex;
```

Comment:
It might be worth splitting this into two, given the writing use-cases are isolated

Reply:
done